### PR TITLE
[Wav2Vec2] Fix `None` loss in docstring for Wav2Vec2ForPreTraining

### PR DESCRIPTION
**Pull request by abdouaziz**
# What does this PR do?

- [ ] It fixes the Nan value returned in the contrastive loss of [Wav2Vec2ForPreTraining](https://huggingface.co/docs/transformers/v4.21.3/en/model_doc/wav2vec2#transformers.Wav2Vec2ForPreTraining)  .
- [ ] Adding sampled_negative_indices as a target allows Wav2Vec2ForPreTraining to calculate the loss.

### CODE REVIEW

File: src/transformers/models/test.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from wav2vec2 import Wav2Vec2 , Wav2Vec2Config
```
Comment (sanchit-gandhi) [inline src/transformers/models/test.py:1]: Looks like there's an erroneous file here!

**Review by sanchit-gandhi — COMMENTED** [review]
Hey @abdouaziz - thanks for looking into the doc strings for Wav2Vec2 pre-training! If you have them to hand, could you possibly copy the results you get before and after the fix just to demonstrate what the issue is? Showing the command you ran to get the NaN in the loss and then the result with the fix would be perfect! Might be faster if you have them ready, otherwise I can try and reproduce!

File: src/transformers/models/test.py (hunk 1 above)
Comment (abdouaziz) [reply src/transformers/models/test.py:1]: Hello @sanchit-gandhi  i join below the code that I took directly on the documentation of Huggingface for the pretraining, but this returns Nan as loss  : 

```py
import torch
from transformers import AutoFeatureExtractor, Wav2Vec2ForPreTraining
from transformers.models.wav2vec2.modeling_wav2vec2 import _compute_mask_indices
from datasets import load_dataset

feature_extractor = AutoFeatureExtractor.from_pretrained("facebook/wav2vec2-base")
model = Wav2Vec2ForPreTraining.from_pretrained("facebook/wav2vec2-base")

ds = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation")
input_values = feature_extractor(ds[0]["audio"]["array"], return_tensors="pt").input_values  # Batch size 1

# compute masked indices
batch_size, raw_sequence_length = input_values.shape
sequence_length = model._get_feat_extract_output_lengths(raw_sequence_length)
mask_time_indices = _compute_mask_indices((batch_size, sequence_length), mask_prob=0.2, mask_length=2)
mask_time_indices = torch.tensor(mask_time_indices, device=input_values.device, dtype=torch.long)

with torch.no_grad():
    outputs = model(input_values, mask_time_indices=mask_time_indices)

# compute cosine similarity between predicted (=projected_states) and target (=projected_quantized_states)
cosine_sim = torch.cosine_similarity(outputs.projected_states, outputs.projected_quantized_states, dim=-1)

# show that cosine similarity is much higher than random
cosine_sim[mask_time_indices.to(torch.bool)].mean() > 0.5

# for contrastive loss training model should be put into train mode
model = model.train()
loss = model(input_values, mask_time_indices=mask_time_indices).loss

print(loss)
```

And here adding the **sample_negative_indices** as target to calculate the loss  : 

```py
import torch
from transformers import AutoFeatureExtractor, Wav2Vec2ForPreTraining
from transformers.models.wav2vec2.modeling_wav2vec2 import _compute_mask_indices , _sample_negative_indices
from datasets import load_dataset

feature_extractor = AutoFeatureExtractor.from_pretrained("facebook/wav2vec2-base")
model = Wav2Vec2ForPreTraining.from_pretrained("facebook/wav2vec2-base")

ds = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation")
input_values = feature_extractor(ds[0]["audio"]["array"], return_tensors="pt").input_values  # Batch size 1

# compute masked indices and negatives inddices
batch_size, raw_sequence_length = input_values.shape

sequence_length = model._get_feat_extract_output_lengths(raw_sequence_length).item()
mask_time_indices = _compute_mask_indices((batch_size, sequence_length), mask_prob=0.2, mask_length=2)
sampled_negative_indices = _sample_negative_indices((batch_size, sequence_length), model.config.num_negatives, mask_time_indices)
mask_time_indices = torch.tensor(mask_time_indices,device=input_values.device, dtype=torch.long)
sampled_negative_indices = torch.tensor(sampled_negative_indices, dtype=torch.long)


with torch.no_grad():
  outputs = model(input_values, mask_time_indices=mask_time_indices)

# compute cosine similarity between predicted (=projected_states) and target (=projected_quantized_states)
cosine_sim = torch.cosine_similarity(outputs.projected_states, outputs.projected_quantized_states, dim=-1)

# show that cosine similarity is much higher than random
cosine_sim[mask_time_indices.to(torch.bool)].mean() > 0.5


# for contrastive loss training model should be put into train mode
model = model.train()
loss = model(input_values, mask_time_indices=mask_time_indices, sampled_negative_indices=sampled_negative_indices).loss
print(loss.item())
```

Comment (abdouaziz) [reply src/transformers/models/test.py:1]: @sanchit-gandhi

Comment (sanchit-gandhi) [reply src/transformers/models/test.py:1]: Hey @abdouaziz - thanks for the code snippet! I now understand the problem we're trying to fix here 🤗

We require `sampled_negative_indices` (distractors) in order to compute the contrastive loss terms. Hence, we only compute the loss if `sampled_negative_indices` is passed to the model. Otherwise, we set all the loss terms to `None` in `Wav2Vec2ForPreTraining`:
https://github.com/huggingface/transformers/blob/693ba2cc79ef46e8323b35f791233ce94f433b5d/src/transformers/models/wav2vec2/modeling_wav2vec2.py#L1480-L1481
So in the first example, `sampled_negative_indices` was omitted, meaning the loss terms were `None`. 

The fix you've implemented passes the `sampled_negative_indices` to the model, meaning the `contrastive_loss` term is computed!

**Comment by sanchit-gandhi** [comment]
Great! Now that we've established the problem:

> loss is `None` as `sampled_negative_indices` is omitted from the args of the model

and verified the fix:

> forward `sampled_negative_indices` to the model

we can touch-up this PR and get it merged!

We'll need to do two things:
1. Remove the erroneous file [src/transformers/models/test.py](https://github.com/huggingface/transformers/pull/18960/files/e94a4d4e7bb108aeba8daafc3de7b58fc5048838#diff-dc9af87a30bbbc682dfaf00796e992367511d96095a2f0a46a8da06a948a7050)
2. Code quality: you can do this simply by running `make style` from the root of the Transformers repo 🤗

Let me know if you have any questions! Cheers!

**Comment by abdouaziz** [comment]
Hello @sanchit-gandhi  Thanks for the help , I am still having some issue even when I make sure that I had installed all the necessary packages I am still having this error . 
Even `pip install black[jupyter]` is installed and when I run `make style`i have the same error here : 
```py

Skipping .ipynb files as Jupyter dependencies are not installed.
You can fix this by running ``pip install black[jupyter]``
would reformat examples/research_projects/lxmert/modeling_frcnn.p
...
```

**Comment by sanchit-gandhi** [comment]
Hey @abdouaziz. Thanks for removing the erroneous file :) Don't worry about the ipynb warning as we've not changed any Python notebooks! Looks like something else is up with `make style` though - we have 567 files changed in this PR! There should only be 2 files changed (wav2vec2 and wav2vec2-conformer).

Could you try the following in turn and check whether the number of files changed drops back down to 2 after each step:
1. Rebasing onto main:
```
git fetch upstream
git rebase upstream/main
```
And then running `make style`.
2. Updating HF doc builder https://pypi.org/project/hf-doc-builder/
```
pip install --upgrade hf-doc-builder
```
And then running `make style`.

If that doesn't fix it we can try some other hacks!

**Comment by sanchit-gandhi** [comment]
Hey @abdouaziz - sorry this has been so arduous! Are you still interested in completing this PR? Feel free to open a new one if you wish and we can go from there!

**Comment by abdouaziz** [comment]
> 
Hello @sanchit-gandhi  , Yes i am interesting to completing this RP , but I am still having  the same issue here the new PR 
https://github.com/huggingface/transformers/pull/19061#issue-1375177421 , 
I am ready for suggestion ??
